src/manager.ts

```diff
@@ -0,0 +1,79 @@
+import { Identity } from '@semaphore-protocol/identity'
+import RLN from './rln'
+import Registry from './registry'
+import Cache, { Status } from './cache'
+import { StrBigInt, VerificationKey } from './types'
+import poseidon from 'poseidon-lite'
+
+export default class RLNManager {
+  identity: Identity
+
+  rlnIdentifier: bigint
+
+  treeDepth: number
+
+  zeroValue: bigint
+
+  rlnInstance: RLN
+
+  registryInstance: Registry
+
+  cacheInstance: Cache
+
+  constructor(
+    wasmFilePath: string,
+    finalZkeyPath: string,
+    verificationKey: VerificationKey,
+    rlnIdentifier: bigint,
+    identity: string,
+    treeDepth?: number,
+  ) {
+    this.rlnIdentifier = rlnIdentifier ? rlnIdentifier : RLN._genIdentifier()
+    this.identity = identity ? new Identity(identity) : new Identity()
+    this.treeDepth = treeDepth ? treeDepth : 20
+    this.zeroValue = poseidon([rlnIdentifier]) // It is highly recommended to use a zero value that is unique to the app and non-zero
+
+    this.rlnInstance = new RLN(wasmFilePath, finalZkeyPath, verificationKey, rlnIdentifier, identity)
+    this.registryInstance = new Registry(this.treeDepth, this.zeroValue)
+    this.cacheInstance = new Cache(this.rlnIdentifier)
+  }
+
+  public async createProof(message: string, epoch: StrBigInt) {
```

Comment:
I like `message` :) It's more explicit to me

Reply:
Ya definitely, "signal" isn't clear at all, but for the internal part of the protocol that other devs won't interact with I think it's fine.